detox/src/devices/drivers/android/exec/ADB.js

```diff
@@ -105,10 +105,10 @@ class ADB {
     const apiLvl = await this.apiLevel(deviceId);
 
     let childProcess;
-    if (apiLvl >= 24) {
+    if (apiLvl >= 23) {
       childProcess = await this.adbCmd(deviceId, `install -r -g -t ${apkPath}`);
     } else {
-      childProcess = await this.adbCmd(deviceId, `install -rg ${apkPath}`);
+      childProcess = await this.adbCmd(deviceId, `install -r -g ${apkPath}`);
```

Comment:
**issue**: This must be `-rg` for API < 23.

Tested in API 21, `-g` option will throw error with:

```
detox[3536] DEBUG: [EXEC_CMD, #8] "/opt/android/platform-tools/adb" -s emulator-14900 shell "getprop ro.build.version.sdk"
detox[3536] TRACE: [EXEC_SUCCESS, #8] 21

detox[3536] DEBUG: [EXEC_CMD, #20] "/opt/android/platform-tools/adb" -s emulator-14900 shell "pm install -r -g /data/local/tmp/detox/Application.apk"
detox[3536] TRACE: [EXEC_SUCCESS, #20] Error: Unknown option: -g

detox[3536] DEBUG: [EXEC_CMD, #22] "/opt/android/platform-tools/adb" -s emulator-14900 shell "pm install -r -g /data/local/tmp/detox/Test.apk"
detox[3536] TRACE: [EXEC_SUCCESS, #22] Error: Unknown option: -g
```

ref: https://github.com/wix/detox/issues/274#issuecomment-356529351
ref: https://github.com/wix/Detox/pull/506/files